komodo/__init__.py

```diff
@@ -1,12 +1,50 @@
 """Komodo software distribution build system."""
 
+import os
+
 from .build import make, pypaths
 from .fetch import fetch
 from .shell import shell, pushd
 from .lint import lint
 from .cleanup import cleanup
 from .maintainer import maintainers
 
+def fixup_python_shebangs(prefix, release):
+    """Fix shebang to $PREFIX/bin/python.
+
+    Some packages installed with pip do not respect target executable, that is,
+    they set as their shebang executable the Python executabl used to build the
+    komodo distribution with instead of the Python executable that komodo
+    deploys.  This breaks the application since the corresponding Python modules
+    won't be picked up correctly.
+
+    For now, we use sed to rewrite the first line in some executables.
+
+    This is a hack that should be fixed at some point.
+
+    """
+    # TODO fix hardcoded fix of ipython, bokeh, ...
+    binpath = os.path.join(args.prefix, args.release, 'root', 'bin')
+    python_ = os.path.join(binpath, 'python')
+
+    bins = []
+    # executables with wrong shebang
+    for bin_ in os.path.walk(binpath):
```

Comment:
I think this as on old - not anymore working syntax? As long as you do not intend to descend the filesystem recursively you could just us `os.listdir( )` - and irrespective  of which method is used I think there is a missing `os.path.join( )` in the `open( )`call.